### PR TITLE
Trigger: show waning when the flow in trigger starts with no hsm  template

### PR DESCRIPTION
**Pull request by AmishaBisht**
## Summary
 Target issue is #3267

while creating the trigger checking if the flow in the trigger is starting with the hsm template or not

### CODE REVIEW

File: lib/glific/triggers.ex [hunk 1]
```diff
@@ -154,19 +154,34 @@ defmodule Glific.Triggers do
...
-    handle_action(flow, attrs, 0)
+  @doc """
+  This function is to check the warning while creating the trigger
+  """
+  @spec check_trigger_warnings(map()) ::
+          {:ok, atom()} | {:warning, String.t()}
+  def check_trigger_warnings(attrs) do
+    case Repo.fetch_by(FlowRevision, %{flow_id: Map.get(attrs, :flow_id), status: "published"}) do
+      {:ok, flow} -> handle_action(flow, attrs, 0)
+    end
```
Comment (mdshamoon) [inline lib/glific/triggers.ex]: Why do we need a case statement here?

File: lib/glific/triggers.ex [hunk 2]
```diff
@@ -154,19 +154,34 @@ defmodule Glific.Triggers do
...
 
-  @spec handle_action(map(), map(), integer()) :: {:ok, Trigger.t()}
+  @spec handle_action(map(), map(), integer()) :: {:ok, atom()} | {:warning, String.t()}
   defp handle_action(flow, attrs, nested_flow_level) do
     action = flow_action(flow)
 
     case action do
       nil ->
-        do_create_trigger(attrs)
+        {:ok, :no_warnings}
```
Comment (mdshamoon) [inline lib/glific/triggers.ex]:
```suggestion
        :ok
```
Why do we need no warnings atom. Just :ok would work

File: assets/gql/triggers/check_warning.gql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#import "./fields.frag.gql"
+
+mutation CheckTriggerWarnings($input: TriggerInput!) {
```
Comment (mdshamoon) [inline assets/gql/triggers/check_warning.gql]:
```suggestion
mutation ValidateTrigger($input: TriggerInput!) {
```
Update the name in all places

File: assets/gql/triggers/check_warning.gql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  checkTriggerWarnings(input: $input) {
+    ...TriggerFields
+
+    ...ErrorFields
```
Comment (mdshamoon) [inline assets/gql/triggers/check_warning.gql]: The response format should be the same as publishFlow API
An attribute for success and errors. We don't need the trigger values here

**Review by mdshamoon — CHANGES_REQUESTED** [review]

**Review by mdshamoon — APPROVED** [review]
Overall looks good. Just check if this sentence can be more user friendly. Looks weird
image

**Comment by AmishaBisht** [comment]
> Overall looks good. Just check if this sentence can be more user friendly image

This warning message came from prod team.

**Pull request merged**